pallets/process-validation/src/lib.rs

```diff
@@ -90,38 +101,141 @@ pub mod pallet {
+        InvalidVersion,
     }
 
     // The pallet's dispatchable functions.
     #[pallet::call]
     impl<T: Config> Pallet<T> {
-        // TODO: implement create_process with correct parameters and impl
         #[pallet::weight(T::WeightInfo::create_process())]
-        pub(super) fn create_process(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
-            // Check it was signed and get the signer
+        pub(super) fn create_process(
+            origin: OriginFor<T>,
+            id: T::ProcessIdentifier,
+            restrictions: Vec<Restriction>,
+        ) -> DispatchResultWithPostInfo {
             T::CreateProcessOrigin::ensure_origin(origin)?;
+            let version: T::ProcessVersion = Pallet::<T>::update_version(id.clone()).unwrap();
+            Pallet::<T>::persist_process(&id, &version, &restrictions)?;
+
+            Self::deposit_event(Event::ProcessCreated(
+                id,
+                version.clone(),
+                restrictions,
+                version == One::one(),
+            ));
 
-            Self::deposit_event(Event::ProcessCreated);
-            Ok(().into())
+            return Ok(().into());
         }
 
-        // TODO: implement disable_process with correct parameters and impl
         #[pallet::weight(T::WeightInfo::disable_process())]
-        pub(super) fn disable_process(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
+        pub(super) fn disable_process(
+            origin: OriginFor<T>,
+            id: T::ProcessIdentifier,
+            version: T::ProcessVersion,
+        ) -> DispatchResultWithPostInfo {
             T::DisableProcessOrigin::ensure_origin(origin)?;
-            Self::deposit_event(Event::ProcessDisabled);
-            Ok(().into())
+            Pallet::<T>::validate_version_and_process(&id, &version)?;
+            Pallet::<T>::set_disabled(&id, &version)?;
+
+            Self::deposit_event(Event::ProcessDisabled(id, version));
+            return Ok(().into());
+        }
+    }
+
+    // helper methods
+    impl<T: Config> Pallet<T> {
+        pub fn get_version(id: &T::ProcessIdentifier) -> T::ProcessVersion {
+            return match <VersionModel<T>>::contains_key(&id) {
+                true => <VersionModel<T>>::get(&id) + One::one(),
+                false => One::one(),
+            };
+        }
+
+        pub fn update_version(id: T::ProcessIdentifier) -> Result<T::ProcessVersion, Error<T>> {
+            let version: T::ProcessVersion = Pallet::<T>::get_version(&id);
+            match version == One::one() {
+                true => <VersionModel<T>>::insert(&id, version.clone()),
+                false => <VersionModel<T>>::mutate(&id, |v| *v = version.clone()),
+            };
+
+            return Ok(version);
+        }
+
+        pub fn persist_process(
+            id: &T::ProcessIdentifier,
+            v: &T::ProcessVersion,
+            r: &Restrictions,
+        ) -> Result<bool, Error<T>> {
```

Comment:
Not going to fail the PR for this but as you only ever return `Some(true)` or `Err(...)` this would likely be better represented as `Result<(), Error<T>>`